src/main/scala/EnsimePlugin.scala

```diff
@@ -362,6 +365,22 @@ object EnsimePlugin extends AutoPlugin {
       artifact = artifactFilter(classifier = Artifact.DocClassifier)
     ).toSet ++ (ensimeUnmanagedJavadocArchives in projectRef).gimme
 
+    val sbv = scalaBinaryVersion.gimme
+    val mainSources = filteredSources(sourcesFor(Compile) ++ sourcesFor(Provided) ++ sourcesFor(Optional), sbv)
+    val testSources = filteredSources(testPhases.flatMap(sourcesFor), sbv)
+    val mainTarget = targetForOpt(Compile).get
+    val testTargets = testPhases.flatMap(targetForOpt).toSet
+    val deps = project.dependencies.map(_.project.project).toSet
+    val mainJars = jarsFor(Compile) ++ unmanagedJarsFor(Compile) ++ jarsFor(Provided) ++ jarsFor(Optional)
```

Comment:
oh, but you don't need to do it this way... you can generate the old config from the new config. You don't need to ask sbt for all the information twice.